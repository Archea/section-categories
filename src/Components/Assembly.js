import React, { Component } from 'react'

class Assembly extends Component {
  handleClick = () => this.props.onSelect(this.props.Assembly)
  render() {
    return (
      <div className="Assembly">
        <div onClick={this.handleClick}>
          ({this.props.Assembly.level}) {this.props.Assembly.code}{' '}
          {this.props.Assembly.description}
        </div>
      </div>
    )
  }

  _pickAssembly = async () => {
    // ... you'll implement this
  }
}

export default Assembly
