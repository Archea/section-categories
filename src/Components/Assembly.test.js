import React from 'react'
import ReactDOM from 'react-dom'
import Assembly from './Assembly'
import renderer from 'react-test-renderer'

describe('Assembly', () => {
  const mocks = {
    Assembly: {
      level: 1,
      code: 'a18',
      description: 'an assembly I guess',
      id: 1,
    },
  }

  test('it at renders and matches snapshot', () => {
    const comp = renderer.create(<Assembly {...mocks} />)
    expect(comp).toBeDefined()
    expect(comp.toJSON()).toMatchSnapshot()
  })
})
