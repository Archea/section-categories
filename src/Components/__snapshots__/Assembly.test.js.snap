// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assembly it at renders and matches snapshot 1`] = `
<div
  className="Assembly"
>
  <div
    onClick={[Function]}
  >
    (
    1
    ) 
    a18
     
    an assembly I guess
  </div>
</div>
`;
